### ISSUE TITLE
User can start, pause, and stop the timer.

### ISSUE THREAD

**Issue by ChinmayPatel** [apigarage/atom-pomodoro-app#2]

**Comment by MajinBui** [comment]
I'll work on this next once the settings branch gets merged.

**Comment by MajinBui** [comment]
Do you want the play function/hotkey to function as a pause/play?  I plan to make the pause and play on one key (like a toggle) and the stop button on a separate key.

**Comment by ChinmayPatel** [comment]
Yeah, totally agree. One Square button for Stop. One button toggles between play and pause.

**Comment by MajinBui** [comment]
For the stop button, my current plan is:
- If the user were to hit stop when the timer is on, the time will stop/pause. 
- If stop is pressed again while the timer is already stopped/paused, the time will reset.
  Is this okay?

**Comment by MajinBui** [comment]
Also, do you want just the hotkeys basically or the actually timer running itself in this branch?  Additionally, what hotkeys would you like?

Also, I made a pull request of what I currently have.
https://github.com/apigarage/atom-pomodoro-app/pull/8

**Comment by apigarage** [comment]
Regarding

> For the stop button, my current plan is:
> - If the user were to hit stop when the timer is on, the time will stop/pause. 
> - If stop is pressed again while the timer is already stopped/paused, the time will reset.
>   Is this okay?

We can think it in terms of states:

**State: Default**
- The stop icon is disabled.
- The play icon is shown.
- When user clicks on stop icon:
  - It does nothing because it is disabled.
- When user clicks on play icon. 
  - Starts the timer. (and makes the time visible)
  - The stop icon is enabled. 
  - Pause Icon is shown.
  - Goes to `Running` state

**State: Running**
- When user clicks on stop icon:
  - Stops the timer.
  - Goes to `Default` state.
- When user clicks on pause icon. 
  - Pauses the timer. 
  - Play Icon is shown.
  - Goes to `Paused` state

**State: Paused**
- When user clicks on stop icon:
  - Stops the timer.
  - Goes to `Default` state.
- When user clicks on pause icon. 
  - Continues the timer. 
  - Pause Icon is shown.
  - Goes to `Running` state

**When the atom is closed**
All hell breaks loose, and we forget everything. On start, we go back to `Default` state.

**Comment by MajinBui** [comment]
Completed.
https://github.com/apigarage/atom-pomodoro-app/pull/8
![timer completed](https://cloud.githubusercontent.com/assets/8998363/10866544/89bcdad2-8009-11e5-8cd9-eb4d565cfe24.gif)
